src/dbos-executor.ts

```diff
@@ -655,9 +659,9 @@ export class DBOSExecutor {
    */
   async flushWorkflowBuffers() {
     if (this.initialized) {
-      await this.systemDatabase.flushWorkflowInputsBuffer();
       await this.flushWorkflowResultBuffer();
       await this.systemDatabase.flushWorkflowStatusBuffer();
+      await this.systemDatabase.flushWorkflowInputsBuffer();
```

Comment:
This could be a misunderstanding on my part... but can items be added to the inputs buffer after the status buffer is flushed, and could those items be added to the DB before the status?  I would expect to see the inputs buffer snapshotted, then the status dumped, then the input snapshot saved, or something like that.

Reply:
This PR forces items be added to the inputs buffer after adding to the status buffer. And no inputs can be added to the DB before workflow status (due to foreign key constraints). If the inputs are failed to exported to the DB (shouldn't happen, but there could be some corner cases that we don't know of), we put them back to the buffer and try to export them again in the next cycle.

Reply:
That much I got... it was about new input records getting created during one of the awaits in the flush of the status buffer.  If we have a problem in the future I would suggest snapshotting the inputs (which is already happening inside flushWorkflowInputsBuffer, then dumping status buffer, then dumping the input snapshot.

Reply:
Yeah we already snapshots the inputs. If we see an issue in the future we might need to combine flushing workflow status and inputs in the same loop.